ERCS/erc-7731.md

```diff
@@ -0,0 +1,182 @@
+---
+eip: 7731
+title: Vulnerability and Exposure Identifier
+description: Establish a specification for creating on-chain unique identifiers for vulnerabilities
+author: john (@0xJohnZW), dimitry (@dimitrysuen), myiwen (@myiwen), msfew (@fewwwww) <msfew@ora.io>
+discussions-to: https://ethereum-magicians.org/t/erc-7731-vulnerability-and-exposure-identifier/20497
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-06-30
+---
+
+## Abstract
+
+This EIP proposes a specification of the identifiers of vulnerabilities and exposures and a rule to specify the unique identifiers for vulnerabilities for disclosure and indexing, establishing a common language for protocol developers and ethical hackers. Indexers can easily track and subscribe to vulnerability updates if the vulnerability database is compatible with this EIP, whether on-chain or off-chain.
+
+## Motivation
+
+When discussing security incidents, they are usually referred to as "Protocol Name + Exploit". Also, there are no unique identifiers for ethical hackers to reference the vulnerabilities, leading to high communication costs between the protocol developers and the ethical hackers. We can unify the terminology across various contexts by providing each vulnerability a unique identifier, as a parallel system to the existing ones such as CVE(Common Vulnerability and Exposure).
+
+This EIP can be useful in the following use cases:
+
+* Both audit and bug reports can reference a specific vulnerability with a unique identifier.
+* Security researchers can subscribe to on-chain events to track the latest vulnerabilities easily.
+
+This EIP can benefit the following parties:
+
+* Ethical hackers can exhibit their research outcomes and establish on-chain reputations.
+* Protocols' security can be enhanced since this EIP encourages open vulnerability reporting.
+* Security researchers can gain more insights on vulnerabilities in more efficient way.
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC-2119.
+
+### Identifier of Vulnerability and Exposure
+
+The identifier of vulnerability and exposure MUST conform to the following format: `Database Prefix + Year + Arbitrary Digits`.
```

Comment:
This is a somewhat ambiguous way of specifying the parts. Are the pluses literal or are they a concatenation operator?

The way you write it later is much better, so how about something like this:

```suggestion
The identifier of vulnerability and exposure SHALL contain three components separated by hyphens (`‐`) in the following order: database prefix, year, and arbitrary digits. An example of this format is `CVE-2024-0097`, where `CVE` is the database prefix, `2024` is the year, and `0097` are the arbitrary digits.
```